graphistry/umap_utils.py

```diff
@@ -214,22 +225,29 @@ def _check_target_is_one_dimensional(self, y: Union[pd.DataFrame, None]):
             return None
 
     def umap_fit(self, X: pd.DataFrame, y: Union[pd.DataFrame, None] = None):
-        self.umap_lazy_init()
         if self._umap is None:
             raise ValueError("UMAP is not initialized")
-
         t = time()
         y = self._check_target_is_one_dimensional(y)
         logger.info('-' * 90)
         logger.info(f"Starting UMAP-ing data of shape {X.shape}")
 
-        self._umap.fit(X, y)
+        is_old = is_old_cuml()
+        if (self.engine == 'cuml' and is_old):
+            import cuml
+            knn = cuml.neighbors.NearestNeighbors(n_neighbors=self.n_neighbors)
+            cc = cuml.UMAP().fit(X, y, knn_graph=knn)
```

Comment:
Use self._umap as its UMAP is already init with proper params, and ensures self._umap is fit when leaving this call

Also, neat on passing in knn_graph here -- maybe worth checking cuml source code on whether they do something special for configuring theirs?